app/jobs/identities_backfill_job.rb

```diff
@@ -0,0 +1,58 @@
+class IdentitiesBackfillJob < ApplicationJob
+  # This is a short-term solution to backfill data requiring a table scan.
+  # This job can be deleted once it's done.
+
+  queue_as :low
+
+  # Let's give us the option to fine-tune this on the fly
+  BATCH_SIZE_KEY = 'IdentitiesBackfillJob.batch_size'.freeze
+  SLICE_SIZE_KEY = 'IdentitiesBackfillJob.slice_size'.freeze
+  CACHE_KEY = 'IdentitiesBackfillJob.position'.freeze
+
+  def perform
+    start_time = Time.zone.now
+    max_id = ServiceProviderIdentity.last.id
+
+    (batch_size / slice_size).times.each do |slice_num|
+      start_id = position + (slice_size * slice_num)
+      next if start_id > max_id
+      params = {
+        min_id: start_id,
+        max_id: start_id + slice_size,
+      }.transform_values { |v| ActiveRecord::Base.connection.quote(v) }
+      sp_query = format(<<~SQL, params)
+        UPDATE identities
+        SET last_consented_at = created_at
+        WHERE id > %{min_id}
+        AND id <= %{max_id}
+        AND deleted_at IS NULL
+        AND last_consented_at IS NULL
+      SQL
+
+      ActiveRecord::Base.connection.execute(sp_query)
+      logger.info "Processed #{slice_size} rows starting at row #{start_id}"
+    end
+
+    elapsed_time = Time.zone.now - start_time
+    logger.info "Finished a full batch of #{batch_size} rows in #{elapsed_time} seconds"
+
+    # If we made it here without error, increment the counter for next time:
+    REDIS_POOL.with { |redis| redis.set(CACHE_KEY, position + batch_size) }
```

Comment:
ok so there's a potential for a bug here with the last batch, where suppose we have 555 records and the last batch starts at 550 and goes for 10, it would "end" at 560. But that's accounting for 5 records that don't exist.

In our case, it's the old (lower id) rows that are missing `last_consented_at` so by the time we get to this case, we're probably fine.

I think the correct thing to do would be to add a `RETURNING id` or something, and grab the max of that.

However, may not be worth the effort